src/devtools/views/Profiler/__tests__/utils-test.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import {
+  prepareExportedProfilingSummaryFromFrontend,
+  parseExportedProfilingDataString,
+} from '../utils';
+
+describe('devtools/views/Profiler/utils', () => {
+  describe('prepareExportedProfilingSummaryFromFrontend', () => {
+    it('works', () => {
+      expect(
+        prepareExportedProfilingSummaryFromFrontend({
+          profilingOperationsByRootID: new Map([
+            [
+              11,
+              [
+                Uint32Array.from([
+                  99,
+                  11,
+                  0 /* this operations array is incomplete compared to a real one */,
+                ]),
+              ],
+            ],
+          ]),
+          profilingSnapshotsByRootID: new Map([
+            [
+              11,
+              {
+                children: [12],
+                displayName: 'DisplayName',
+                id: 11,
+                key: 'key',
+              },
+            ],
+          ]),
+          rendererID: 99,
+          rootID: 11,
+        })
+      ).toMatchSnapshot();
+    });
+  });
+
+  describe('parseExportedProfilingDataString', () => {
+    it('works', () => {
+      expect(
+        // This is a recording of `shells/dev/index.html`
+        parseExportedProfilingDataString(`
```

Comment:
Wouldn't it be sufficient (and easier to read) to just define some profiling data, export, re-import, and compare to original? What's the function of this hard-coded blob of JSON?

Reply:
> > P.S. I'd love to not include huge JSONs and snapshots, but I'm not sure how to get a smaller profiling session recording.
> 
> What's "huge" about this data? Be specific. I think it's pretty streamlined to enable us to lazily compute the bits that we need to make the frontend/UI work.

Sorry for not making it very clear, I meant this JSON in the test.

> Wouldn't it be sufficient (and easier to read) to just define some profiling data, export, re-import, and compare to original? What's the function of this hard-coded blob of JSON?

Sure (although I wanted this test to not depend on the whole Store/Bridge infra), by "just define some profiling data" what do you mean specifically?

Reply:
> Sorry for not making it very clear, I meant this JSON in the test.

Oh! That makes a lot more sense now. Thanks for clarifying 😄 

> Sure (although I wanted this test to not depend on the whole Store/Bridge infra), by "just define some profiling data" what do you mean specifically?

Yeah, sorry. My original comment was not very clear. I was just thinking we could combine the two above tests into a single one that only has one hard-coded source of data.

1. Start with a hard-coded object of profiler data (this part is still yuck)
1. Serialize it using `prepareExportedProfilingSummaryFromFrontend` then `JSON.stringify`
1. Reconstruct it using `parseExportedProfilingDataV1`
1. Expect to be shallow-equal to the original

Would probably be better still if we could generate *real* profiling data though, as hard-coded mocks can get stale easily.

Reply:
Oops accidentally resolved, github made "reply" not a button. Unresolved now.

Reply:
As I remember from last night that these two things `prepareExportedProfilingSummaryFromFrontend` and `parseExportedProfilingDataV1` aren't type-compatible. But I need to take a fresh look.

Reply:
Conceptually, all I'm saying is that it would be nice if we had a single test that verified that we could start with some known profiling data<sup>1</sup>, serialize it, then deserialize it and end up with the same data.

<sup>1</sup> Preferably *real* data rather than hard-coded data.

Reply:
Yes, I agree. But this would require spinning up the whole infra, rendering the app, profiling it, exporting, importing. I agree such a test would be more valuable than this one.

Reply:
I don't think I understand. We are essentially already doing most of this for our tests, and it's very valuable.

Reply:
> > Yes, I agree. But this would require spinning up the whole infra, rendering the app, profiling it, exporting, importing. I agree such a test would be more valuable than this one.
> 
> I don't think I understand. We are essentially already doing most of this for our tests, and it's very valuable.

Sorry for the embedded contradiction in my line of thought. I was just complaining, but agreed that the test should be changed.

Reply:
No worries :)